test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndChannelLeakTest.java

```diff
@@ -177,43 +177,40 @@ public void testDetectChannelLeakMultiReader() throws Exception {
         ConnectionFactoryImpl connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
         @Cleanup
         ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
-
-        assertEquals(0, connectionFactory.getActiveChannelCount());
+        int expectedChannelCount = 0; // open socket count.
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
         
-        //Create a writer.
+        //Create a writer and write an event.
         @Cleanup
         EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM_NAME, new JavaSerializer<>(),
                 EventWriterConfig.builder().build());
-        
-        assertEquals(0, connectionFactory.getActiveChannelCount());
-        //Write an event.
         writer.writeEvent("0", "zero").get();
-        assertEquals(1, connectionFactory.getActiveChannelCount());
+
+        expectedChannelCount += 1; // connection to segment 0.
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
         
         @Cleanup
         ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
                 connectionFactory);
-        assertEquals(1, connectionFactory.getActiveChannelCount());
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount()); // no changes expected.
       
         groupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().disableAutomaticCheckpoints()
                 .stream(Stream.of(SCOPE, STREAM_NAME)).build());
-        //create a reader.
+
+        //create a reader and read an event.
         @Cleanup
         EventStreamReader<String> reader1 = clientFactory.createReader("readerId1", READER_GROUP, serializer,
                 ReaderConfig.builder().build());
-
-        //Total 5 sockets are open at this point : Writer has 1 connection to segment 0 of stream +
-        //Reader has 4 connections to the reader group (read, write, metadata, and conditionalUpdates)
-        assertEquals(5, connectionFactory.getActiveChannelCount());
-        int channelCount = 5;
-
-        //Read an event.
+        //Creating a reader spawns a revisioned stream client which opens 4 sockets ( read, write, metadataClient and conditionalUpdates).
+        expectedChannelCount += 4;
```

Comment:
An assertion was removed here that should be still present.

Reply:
Removed the assertion here as the `operationComplete` callback (ref: https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java#L157) might or might not be completed when the check for activeChannel count for a value of 5 is performed.

Modified the test to ensure that we check the channel count when we are absolutely sure that connection has been established.